test/format/format.sh

```diff
@@ -238,32 +295,59 @@ resolve()
 				success=$(($success + 1))
 				verbose "$name: job in $dir successfully completed"
 			else
-				echo "$name: failure status reported" > $dir/reported
-				failure=$(($failure + 1))
 				echo "$name: job in $dir failed abort/recovery testing"
+				report_failure $dir
 			fi
 			continue
 		}
 
-		# Discard jobs where the timer went off.
-		grep 'caught signal' $log > /dev/null && {
-			rm -rf $dir $log
-			verbose "$name: job in $dir aborted"
+		# Check for the library abort message, or an error from format.
+		grep -E 'aborting WiredTiger library|run FAILED' $log > /dev/null && {
+			report_failure $dir
 			continue
 		}
 
-		# Report failures.
-		# Check for the library abort message, or an error from format.
-		grep -E 'aborting WiredTiger library|run FAILED' $log > /dev/null && {
-			echo "$name: job in $dir failed"
-			echo "$name: $dir log:"
-			echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
-			sed 's/^/    /' < $log
-			echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
-
-			echo "$name: failure status reported" > $dir/reported
-			failure=$(($failure + 1))
+		# There's some chance we just dropped core. We have the exit status of the process,
+		# but there's no way to be sure. There are reasons the process' exit status looks
+		# like a core dump was created (format deliberately causes a segfault in the case
+		# of abort/recovery testing, and does work that can often segfault in the case of a
+		# snapshot-isolation mismatch failure), but those cases have already been handled,
+		# format is responsible for logging a failure before the core can happen. If the
+		# process exited with a likely failure, call it a failure.
+		signame=""
+		case $eret in
+		$((128 + 3)))
+			signame="SIGQUIT";;
+		$((128 + 4)))
+			signame="SIGILL";;
+		$((128 + 6)))
+			signame="SIGABRT";;
+		$((128 + 7)))
+			signame="SIGBUS";;
+		$((128 + 8)))
+			signame="SIGFPE";;
+		$((128 + 11)))
+			signame="SIGSEGV";;
+		$((128 + 24)))
+			signame="SIGXCPU";;
+		$((128 + 25)))
+			signame="SIGXFSZ";;
+		$((128 + 31)))
+			signame="SIGSYS";;
+		esac
+		[[ ! -z $signame ]] && {
+			(echo
+			 echo "$name: job in $dir killed with signal $signame"
+			 echo "$name: there may be a core dump associated with this failure"
+			 echo) >> $log
+
+			echo "$name: job in $dir killed with signal $signame"
+			echo "$name: there may be a core dump associated with this failure"
```

Comment:
Could you look in `$dir` for `*core*` and tell the user the core dump file name?

Reply:
No, because Linux let's you name the core dump anything you want.